diff --git a/configure.ac b/configure.ac
index 49f2cbc7..00b094fd 100644
--- a/configure.ac
+++ b/configure.ac
@@ -310,11 +310,11 @@ Either configure with --disable-plugins or resolve this error to build HTSlib.])
                      [Platform-dependent plugin filename extension.])
 fi
 
-AC_SEARCH_LIBS([log], [m], [],
+AC_CHECK_DECL([log], [],
   [MSG_ERROR([log() not found
 
 HTSLIB requires a working floating-point math library.
-FAILED.  This error must be resolved in order to build HTSlib successfully.])])
+FAILED.  This error must be resolved in order to build HTSlib successfully.])], [[#include <math.h>]])
 
 zlib_devel=ok
 dnl Set a trivial non-empty INCLUDES to avoid excess default includes tests
