diff --git a/hts_internal.h b/hts_internal.h
index 52f29e6..5d880c8 100644
--- a/hts_internal.h
+++ b/hts_internal.h
@@ -123,7 +123,7 @@ const char *hts_plugin_path(void);
  */
 int bgzf_idx_push(BGZF *fp, hts_idx_t *hidx, int tid, hts_pos_t beg, hts_pos_t end, uint64_t offset, int is_mapped);
 
-static inline int find_file_extension(const char *fn, char ext_out[static HTS_MAX_EXT_LEN])
+static inline int find_file_extension(const char *fn, char ext_out[HTS_MAX_EXT_LEN])
 {
     const char *delim = fn ? strstr(fn, HTS_IDX_DELIM) : NULL, *ext;
     if (!fn) return -1;
